### DISCUSSION TITLE
How to cancel a request onBeforeSend

### DISCUSSION THREAD

**Discussion by ajomuch92** [pocketbase/pocketbase#5090] [Q&A]
I'm trying to store some information locally, so I want to check the internet connection onBeforeSend method for pocketbase instance to get the info saved locally when there is no internet. Is there a way to do that? Or there is a global method called when an error happens to return the info saved locally?

**Comment by ganigeorgiev** [comment]
You can provide your own custom `fetch` function with the `options` but note that the custom `fetch` must return a Promise that resolves to a [`Response` object](https://developer.mozilla.org/en-US/docs/Web/API/Response). Something like:

```js
pb.beforeSend = function (url, options) {
    options.fetch = async (url, config) => {
        if (window.navigator.onLine) {
            return globalThis.fetch(url, config)
        }

        // your own custom local "fetch" (must return Response obj)...
    }

    return { url, options };
};
```

**Reply by ajomuch92** [reply]
It says this
![image](https://github.com/pocketbase/pocketbase/assets/32050436/effcea56-3069-41f3-93d6-9a2d7f263e17)

**Reply by ganigeorgiev** [reply]
Use `options.fetch` not `options.request`.

Alternatively if you want to attach other fields you can simply type assert it to `any` to get around the TS errors.
